packages/dataviews/src/dataviews-layouts/picker-table/index.tsx

```diff
@@ -0,0 +1,490 @@
+}: ViewPickerTableProps< Item > ) {
+	const { containerRef } = useContext( DataViewsContext );
+	const headerMenuRefs = useRef<
+		Map< string, { node: HTMLButtonElement; fallback: string } >
+	>( new Map() );
+	const headerMenuToFocusRef = useRef< HTMLButtonElement >();
+	const [ nextHeaderMenuToFocus, setNextHeaderMenuToFocus ] =
+		useState< HTMLButtonElement >();
+	const hasBulkActions = useSomeItemHasAPossibleBulkAction( actions, data );
+
+	useEffect( () => {
+		if ( headerMenuToFocusRef.current ) {
+			headerMenuToFocusRef.current.focus();
+			headerMenuToFocusRef.current = undefined;
+		}
+	} );
+
+	const tableNoticeId = useId();
+
+	// Note: We don't use isHorizontalScrollEnd for pickers as we don't have sticky actions column
+	useIsHorizontalScrollEnd( {
+		scrollContainerRef: containerRef,
+		enabled: false,
+	} );
+
+	if ( nextHeaderMenuToFocus ) {
+		// If we need to force focus, we short-circuit rendering here
+		// to prevent any additional work while we handle that.
+		// Clearing out the focus directive is necessary to make sure
+		// future renders don't cause unexpected focus jumps.
+		headerMenuToFocusRef.current = nextHeaderMenuToFocus;
+		setNextHeaderMenuToFocus( undefined );
+		return;
+	}
+
+	const onHide = ( field: NormalizedField< Item > ) => {
+		const hidden = headerMenuRefs.current.get( field.id );
+		const fallback = hidden
+			? headerMenuRefs.current.get( hidden.fallback )
+			: undefined;
+		setNextHeaderMenuToFocus( fallback?.node );
+	};
+
+	const hasData = !! data?.length;
+
+	const titleField = fields.find( ( field ) => field.id === view.titleField );
+	const mediaField = fields.find( ( field ) => field.id === view.mediaField );
+	const descriptionField = fields.find(
+		( field ) => field.id === view.descriptionField
+	);
+
+	const groupField = view.groupByField
+		? fields.find( ( f ) => f.id === view.groupByField )
+		: null;
+	const dataByGroup = groupField ? getDataByGroup( data, groupField ) : null;
+	const { showTitle = true, showMedia = true, showDescription = true } = view;
+	const hasPrimaryColumn =
+		( titleField && showTitle ) ||
+		( mediaField && showMedia ) ||
+		( descriptionField && showDescription );
+	const columns = view.fields ?? [];
+	const headerMenuRef =
+		( column: string, index: number ) => ( node: HTMLButtonElement ) => {
+			if ( node ) {
+				headerMenuRefs.current.set( column, {
+					node,
+					fallback: columns[ index > 0 ? index - 1 : 1 ],
+				} );
+			} else {
+				headerMenuRefs.current.delete( column );
+			}
+		};
+	const isInfiniteScroll = view.infiniteScrollEnabled && ! dataByGroup;
+
+	return (
+		<>
+			<table
+				className={ clsx( 'dataviews-view-table', className, {
+					[ `has-${ view.layout?.density }-density` ]:
+						view.layout?.density &&
+						[ 'compact', 'comfortable' ].includes(
+							view.layout.density
+						),
+				} ) }
+				aria-busy={ isLoading }
+				aria-describedby={ tableNoticeId }
+				role={ isInfiniteScroll ? 'feed' : undefined }
```

Comment:
One of my concerns over `table` is whether the aria semantics are right for a picker.

For grid and list I think it's easy to use the `listbox` role and it works well, but for table it's a bit harder to find the right fit because table already has its own role. Maybe there's a way to make it work. 🤔 

Perhaps `grid` is a better fit than the default table roles, as I believe it's supposed to support `aria-selected` for rows - https://www.w3.org/WAI/ARIA/apg/patterns/grid/.

Reply:
I think infinite scroll should always be a `feed` role regardless. whether that's compatible with the picker semantics I don't know 😅

Reply:
Is it worth updating the README as well?

https://github.com/WordPress/gutenberg/blob/try/data-picker-table-view/packages/dataviews/README.md#dataviewspicker

> The items in the view are rendered using the listbox and option aria roles.

Asking coz I don't know 😄

Reply:
Perhaps it would work.

So I guess the idea would be to make the table the `listbox` and then the table rows become `option`s, but then for table all the cells also have implicit aria roles, and that's the part that I'm not sure what to do:
<img width="404" height="47" alt="Screenshot 2025-11-05 at 1 15 26 pm" src="https://github.com/user-attachments/assets/e9dbd92d-bbc2-4960-9d75-fe559caa37d3" />

I don't think `cell` roles work within `option` roles.

Similar situation with the header:
<img width="196" height="64" alt="Screenshot 2025-11-05 at 1 16 50 pm" src="https://github.com/user-attachments/assets/02631957-598d-4c13-af18-749eca5d01d3" />

I think you end up losing the header relationship with the cells if it's switched to `listbox`.

Reply:
Ok potentially controversial question: do we really need a header? It's unlikely the picker table will have multiple rows, aside from primary and checkbox? 

Or rather, thinking strictly of the non-visual interaction that the aria roles cater to, is it helpful at all to have table semantics in a widget that is designed specifically for selection? If it isn't, we could just remove the table-related roles altogether.

Reply:
Do you mean it's unlikely to have multiple columns? I agree, I think most pickers should start with a minimal set of fields but allow user to add more to distinguish items if they need to.

Removing the table semantics is what I was thinking as well. The `grid` example here sets `role="presentation"` on rows, so it could be similar for listbox:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/grid_role

Reply:
And I guess you could also set presentation role on the header too, so that it's still visible. If we need to add headers for screenreaders we could also try visually hidden text in the cells.

Reply:
> Do you mean it's unlikely to have multiple columns?

Yes sorry, columns, not rows!

OK let's try a listbox and removing table semantics. We can always revisit later if we find it doesn't work.

Reply:
Oh, I was actually wondering if folks would use multiple columns to be able to sort by different columns easily. Would that still be possible in a listbox?

Reply:
Yeah the dropdowns work. I'm not sure how clear it would be for a screen reader user that e.g. the "title" menu pop-up button hides sorting options for that column, but that's probably also the case in the table layout.